project/Build.scala

```diff
@@ -721,14 +749,108 @@ object Build {
       ) ++ (
           parallelCollectionsDependencies(scalaVersion.value)
       ),
+
+      resourceGenerators in Compile += Def.task {
+        val baseResourceDir = (resourceManaged in Compile).value
+        val resourceDir = baseResourceDir / "org/scalajs/linker/backend/emitter"
+
+        /* Awful hack to execute `products in (linkerPrivateLibrary, Compile)`
+         * as if we were in ++2.12.8, regardless of the current version of the
+         * linker and other projects.
+         */
+        val privateLibProducts = {
+          val s = state.value
+          val relevantProjects = List(compiler, library, libraryAux, scalalib,
+              javalanglib, javalib, linkerPrivateLibrary)
+          val setVersionSettings = relevantProjects.map { p =>
+            scalaVersion in p := "2.12.8"
+          }
+          val newState =
+            appendWithSession(Project.extract(s), setVersionSettings, s)
+          val extracted = Project.extract(newState)
+          val (_, privateLibProducts) =
+            extracted.runTask(products in (linkerPrivateLibrary, Compile), newState)
+          privateLibProducts
+        }
+
+        IO.createDirectory(resourceDir)
+        val copyPairs = for {
+          f <- (privateLibProducts ** "*.sjsir").get
+        } yield {
+          f -> resourceDir / f.getName
+        }
+        IO.copy(copyPairs, overwrite = false, preserveLastModified = true).toSeq
+      }.taskValue,
+
       fork in Test := true
   ).dependsOn(irProject, io, logging)
 
+  // Copy-pasted `Extracted.appendWithSession` from sbt, available only in sbt 1.1.
+  private def appendWithSession(extracted: Extracted, settings: Seq[Setting[_]],
+      state: State): State = {
+    appendImpl(extracted, settings, state, extracted.session.mergeSettings)
+  }
+
+  private[this] def appendImpl(extracted: Extracted, settings: Seq[Setting[_]],
+      state: State, sessionSettings: Seq[Setting[_]]): State = {
+    import extracted.{currentRef, rootProject, session, showKey, structure}
+    val appendSettings = Load.transformSettings(Load.projectScope(currentRef),
+        currentRef.build, rootProject, settings)
+    val newStructure =
+      Load.reapply(sessionSettings ++ appendSettings, structure)
+    Project.setProject(session, newStructure, state)
+  }
+
   lazy val linkerJS: Project = (project in file("linker/js")).enablePlugins(
       MyScalaJSPlugin
   ).settings(
       commonLinkerSettings,
       crossVersion := ScalaJSCrossVersion.binary,
+
+      sourceGenerators in Compile += Def.task {
+        val dir = (sourceManaged in Compile).value
+        val privateLibProducts = (products in (linkerPrivateLibrary, Compile)).value
```

Comment:
Shouldn't this also take the 2.12.8 RuntimeLong for consistency at least?

Reply:
Yes, indeed. That would be better.